src/test/java/org/pesmypetcare/webservice/dao/WeekTrainingDaoTest.java

```diff
@@ -0,0 +1,380 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.pesmypetcare.webservice.entity.WeekTrainingEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class WeekTrainingDaoTest {
+    private static List<Map<String, Object>> weekTrainingList;
+    private static WeekTrainingEntity weekTrainingEntity;
+    private static String date;
+    private static String date2;
+    private static String owner;
+    private static String petName;
+    private static Double value;
+    private static String field;
+    private static final String USERS_KEY = "users";
+    private static final String PETS_KEY = "pets";
+    private static final String WEEKTRAININGS_KEY = "weekTrainings";
+    private final String EXCECUTION_EXC_MSG = "Should throw DatabaseAccessException when ExecutionException received";
+    private final String INTERRUPTED_EXC_MSG = "Should throw DatabaseAccessException when InterruptedException "
+        + "received";
+    private final String DOCUMENT_NOT_EXISTS_EXC_MSG = "Should throw DatabaseAccessException when WeekTraining document"
+        + " doesn't exist";
+
+    @Mock
+    private Firestore db;
+    @Mock
+    private CollectionReference usersRef;
+    @Mock
+    private DocumentReference ownerRef;
+    @Mock
+    private CollectionReference petsRef;
+    @Mock
+    private DocumentReference petRef;
+    @Mock
+    private CollectionReference weekTrainingsRef;
+    @Mock
+    private DocumentReference weekTrainingRef;
+    @Mock
+    private ApiFuture<QuerySnapshot> futureQuery;
+    @Mock
+    private QuerySnapshot querySnapshot;
+    @Mock
+    private ApiFuture<DocumentSnapshot> futureDocument;
+    @Mock
+    private DocumentSnapshot documentSnapshot;
+    @Mock
+    private List<QueryDocumentSnapshot> weekTrainingsDocuments;
+    @Mock
+    private Iterator<QueryDocumentSnapshot> it;
+
+    @InjectMocks
+    private WeekTrainingDao weekTrainingDao = new WeekTrainingDaoImpl();
+
+    @BeforeAll
+    public static void setUp() {
+        weekTrainingList = new ArrayList<>();
+        weekTrainingEntity = new WeekTrainingEntity();
+        date = "2019-10-22T00:47:00";
+        date2 = "2021-10-22T00:47:00";
+        owner = "PericoDeLosPalotes";
+        petName = "TupoJohn";
+        field = "weekTrainingValue";
+        value = 9.0 / 2.0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['9.0' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5400494)